### ISSUE TITLE
Imread format: unclear what are allowed values

### ISSUE THREAD

**Issue by nscozzaro** [matplotlib/matplotlib#19907]
In this description, the documentation provides no instruction on allowed values. This information is necessary in order to properly use the function. For example for a jpeg do you add format='.JPEG', 'JPEG', 'jpeg', 'JPG' etc...
https://github.com/matplotlib/matplotlib/blob/7ce970e1c12824cebc47223e497fa92d1bbaedc2/lib/matplotlib/image.py#L1437-L1440

**Comment by jklymak** [comment]
We could probably just point to `PIL.Image.open` here.  If I understand the PIL docs this list of formats depends on your installed libraries, so I'm not sure we can or should explicitly list the allowed formats.  Also, these things are pretty standard.

**Comment by tacaswell** [comment]
We special case png to a known fast path (which is probably an artifact of us formerly owning a wrapping of lib png) and some special casing of urls, but otherwise we fall back to `PIL.Image.open`.  That is probably worth noting (and directing people to please do that directly and not use `imread`!).

I have several concerns about documenting supported extension here.  As @jklymak points out the list may depend on exactly how PIL is compiled or what other libraries are installed, we do not want to document any normalization / aliasing that `PIL` may do (as if they change that then our docs would be wrong!), and finally the extension on a filename is at best a hint of what is actually in the file (I can name a png `blah_blah.jpeg`).  Almost all file formats have some "magic bytes" at the start to identify them (see https://en.wikipedia.org/wiki/List_of_file_signatures), I expect that PIL has some fallback paths where it reads the first couple of bytes and then dispatches based on that.

----

TLDR: :+1: on linking to `PIL.Image.open` :-1: on listing anything about file extensions.

**Comment by nscozzaro** [comment]
> please do that directly and not use imread
Are you saying you think imread should be removed from matplotlib? That actually would make more sense to me, if it's essentially just a wrapper.

**Comment by jklymak** [comment]
Yes it is a vestige of when PIL was a difficult dependency/didn't exist.  Also Matplotlib has a legacy in matlab, which has imread. I think there is general consensus that we should try to stick to just plotting and not emulate everything matlab does.   OTOH deprecations are painful particularly for simple wrappers, so...

**Comment by nscozzaro** [comment]
Just for context of how I ended up frustrated with the docs, I ran into this problem:
<img src="https://user-images.githubusercontent.com/15185105/114081662-b1f7f880-9861-11eb-9cee-5728411e0ee3.png" width="548">

And solved it like this:
```
from PIL import Image
import requests
import matplotlib.pyplot as plt
url = small_df.at[0, 'image_url']
response = requests.get(url, stream=True)
img = Image.open(response.raw)
plt.imshow(img)
plt.show()
```
I didn't know at the time that Matplotlib was using PIL behind the scenes, but that was my solution anyways.

**Comment by tacaswell** [comment]
> Are you saying you think imread should be removed from matplotlib? 

If we were starting from scratch I would be against putting it in, but it has been there from 2004 and removing it would break a lot of people for mostly aesthetic reasons...so it stays.
